org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/MavenModelManager.java

```diff
@@ -106,13 +105,26 @@ public org.apache.maven.model.Model readMavenModel(InputStream reader) throws Co
   }
 
   /**
-   * Read the model from the provided pom file (might not exits) pointer
+   * Shortcut for {@link #readMavenModel(File, IProgressMonitor) with 2nd argument being {@code null}.
    * 
-   * @param pomFile the file pointer
-   * @return a maven model, or <code>null</code> if the pointer do not point to any valid maven directory
-   * @throws CoreException if the file points to a valid maven directory but the pom could not be read
+   * @deprecated Use {@link #readMavenModel(File, IProgressMonitor)} instead.
    */
+  @Deprecated
   public org.apache.maven.model.Model readMavenModel(File pomFile) throws CoreException {
```

Comment:
@laeubi Are you suggesting to keep this as is (i.e. raw model) and rather introduce a new method for returning the effective model? The only consumer for the raw model I see within m2e is probably org.eclipse.m2e.tests.common/src/org/eclipse/m2e/tests/common/AbstractMavenProjectTestCase.java...